### PR TITLE
Release 3.4.0

### PR DESCRIPTION
**Pull request by percona-csalguero**
Release 3.4.0

- Fixed bug   PT-1218: pt-stalk ominous open_tables function
- Fixed bug   PT-1336: pt-stalk removes user's files from the destination directory
- Fixed bug   PT-1398: pt-stalk get wrong mysqld pid when the host installed multi mysql instance
- Improvement PT-1417: Inconsistent creation of toolkit tables
- Fixed bug   PT-1627: pt-mysql-summary doesn't verify which version of jemalloc is in use
- Fixed bug   PT-1747: pt-online-schema-change: metadata lock can break database for rebuild_constraints
- Improvement PT-1800: PTDEBUG=1 exposes passwords
- Fixed bug PT-1887: pt-diskstat is not working for new kernels
- Fixed bug   PT-1900: pt-query-digest not hiding parameter properly sometimes when parameter=binary
- Improvement PT-1940: ptsoc dropswap with mysql8: revise rejection (Thanks duxthefux)
- Fixed bug   PT-1953: pt-summary typo: Memory mamagement.
- Fixed bug   PT-1959: go part of the toolkit still has the version 3.3.0
- Fixed bug   PT-1965: pt-stalk --mysql-only doesn't collect mysqladmin outputs (Thanks Sergey Kuzmichev)
- Fixed bug   PT-1966: Test no_drop_no_swap for the pt-online-schema-change is broken (Thanks Tsubasa Tanaka)
- Fixed bug   PT-1974: Support fingerprinting for --print in pt-kill
- New Tool    PT-1978: Add reporting on unused/redundant indexes for MongoDB by pt-mongodb-summary
- Improvement PT-1979: Add gathering of admin parameters for MongoDB by pt-mongodb-summary
- Fixed bug   PT-1983: pt-summary missing one DIMM
- Admin task  PT-2009: Update Go Lang dependencies
- Admin task  PT-2011: Make build reproducible
- Fixed bug   PT-2016: pt-table-checksum fails to build replace query when table lack a primary key
- Fixed bug   PT-2023: pt-upgrade Error: Wide character in print
- Admin task  PT-2029: Update CONTRIBUTING.md
- Improvement PT-2037: Add option --skip-mysql or --system-only for pt-stalk

### CODE REVIEW

File: src/go/pt-mongodb-index-check/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# pt-mongodb-index-check
```
Comment (denisok) [inline src/go/pt-mongodb-index-check/README.md]: is this tool here intentionally or it should be in a separate PR ?

**Comment by percona-csalguero** [comment]
Both. There is a different pr which is in review

File: src/go/pt-mongodb-index-check/README.md (hunk 1 above)
Comment (percona-csalguero) [reply src/go/pt-mongodb-index-check/README.md]: There is another PR but it is in review. https://github.com/percona/percona-toolkit/pull/537
I wanted to prepare the release branch including this new tool.

**Comment by rnovikovP** [comment]
I don't think we need the Admin task mentioned. 
 LEt's list the  items in : 
1) a new  tool 
2) Improvements 
3) fixed bugs

File: Changelog [hunk 2]
```diff
@@ -1,21 +1,31 @@
 Changelog for Percona Toolkit
 
-v3.3.2 WIP release date not set yet
+v3.4.0
 
+ * Fixed bug   PT-1218: pt-stalk ominous open_tables function
+ * Fixed bug   PT-1336: pt-stalk removes user's files from the destination directory
+ * Fixed bug   PT-1398: pt-stalk get wrong mysqld pid when the host installed multi mysql instance
```
Comment (rasika-chivate) [inline Changelog]:
```suggestion
 * Fixed bug   PT-1398: pt-stalk gets the incorrect mysqld pid when the host installed a multi MySQL instance
```

File: Changelog [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
  * Improvement PT-1417: Inconsistent creation of toolkit tables
  * Fixed bug   PT-1627:	pt-mysql-summary doesn't verify which version of jemalloc is in use
```
Comment (rasika-chivate) [inline Changelog:14]:
```suggestion
 * Fixed bug   PT-1627:	pt-mysql-summary doesn't verify the version of jemalloc in use
```

File: Changelog [hunk 4]
```diff
@@ -1,21 +1,31 @@
...
  * Fixed bug   PT-1747: pt-online-schema-change: metadata lock can break database for rebuild_constraints
  * Improvement PT-1800: PTDEBUG=1 exposes passwords
```
Comment (rasika-chivate) [inline Changelog]:
```suggestion
 * Improvement PT-1800: The environment variable PTDEBUG=1 exposes the passwords
```

File: Changelog [hunk 5]
```diff
@@ -1,21 +1,31 @@
...
+ * Improvement PT-1887: pt-diskstat is not working for new kernels
  * Fixed bug   PT-1900:	pt-query-digest not hiding parameter properly sometimes when parameter=binary
```
Comment (rasika-chivate) [inline Changelog]:
```suggestion
 * Fixed bug PT-1900:   At times, pt-query-digest does not hide the parameters properly when parameter=binary
```

File: Changelog [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
  * Improvement PT-1940:	ptsoc dropswap with mysql8: revise rejection (Thanks duxthefux)
```
Comment (rasika-chivate) [inline Changelog]:
```suggestion
 * Improvement PT-1940:	ptsoc dropswap method that was rejected for Mysql 8 has been fixed with Mysql 8.0.14
```

File: Changelog [hunk 7]
```diff
@@ -1,21 +1,31 @@
...
  * Fixed bug   PT-1953: pt-summary typo: Memory mamagement.
  * Fixed bug   PT-1959: go part of the toolkit still has the version 3.3.0
  * Fixed bug   PT-1965: pt-stalk --mysql-only doesn't collect mysqladmin outputs (Thanks Sergey Kuzmichev)
```
Comment (rasika-chivate) [inline Changelog]:
```suggestion
 * Fixed bug PT-1965: pt-stalk --mysql-only doesn't collect mysqladmin output
```

File: Changelog [hunk 8]
```diff
@@ -1,21 +1,31 @@
...
  * Fixed bug   PT-1966: Test no_drop_no_swap for the pt-online-schema-change is broken (Thanks Tsubasa Tanaka)
```
Comment (rasika-chivate) [inline Changelog:21]:
```suggestion
 * Fixed bug   PT-1966: Test no_drop_no_swap for the pt-online-schema-change is broken
```

File: Changelog [hunk 9]
```diff
@@ -1,21 +1,31 @@
...
  * Fixed bug   PT-1974: Support fingerprinting for --print in pt-kill
+ * New Tool    PT-1978: Add reporting on unused/redundant indexes for MongoDB by pt-mongodb-summary
+ * Improvement PT-1979: Add gathering of admin parameters for MongoDB by pt-mongodb-summary
  * Fixed bug   PT-1983:	pt-summary missing one DIMM	
  * Admin task  PT-2009: Update Go Lang dependencies
  * Admin task  PT-2011:	Make build reproducible
+ * Fixed bug   PT-2016: pt-table-checksum fails to build replace query when table lack a primary key
```
Comment (rasika-chivate) [inline Changelog]:
```suggestion
 * Fixed bug   PT-2016: pt-table-checksum fails to build replace query when table lacks primary key
```

**Comment by percona-csalguero** [comment]
> I don't think we need the Admin task mentioned. LEt's list the items in :
> 
> 1. a new  tool
> 2. Improvements
> 3. fixed bugs

:+1:

**Review by nikita-b — APPROVED** [review]

**Review by rasika-chivate — CHANGES_REQUESTED** [review]

File: Changelog (hunk 7 above)
Comment (percona-csalguero) [reply Changelog]: Usually we mention 3rd party collaborators.

**Review by denisok — APPROVED** [review]

**Review by rasika-chivate — APPROVED** [review]

**Comment by rnovikovP** [comment]
Are there separate Release notes, or we'll use  Changelog? if changelog - then only ordering request from me https://github.com/percona/percona-toolkit/pull/539#issuecomment-1089141926

**Review by rnovikovP — CHANGES_REQUESTED** [review]
see https://github.com/percona/percona-toolkit/pull/539#issuecomment-1131223887

**Comment by percona-csalguero** [comment]
> see [#539 (comment)](https://github.com/percona/percona-toolkit/pull/539#issuecomment-1131223887)

Usually release notes are a copy of the changelog.
I am going to fix the ordering in the issues list.

**Comment by xjunior** [comment]
Hello, any news about this release? We're having to patch our percona-toolkit with https://github.com/percona/percona-toolkit/commit/fe269e5c948b995b4d1456adc30c8d84ca048325 to be able to run some migrations.

**Comment by benlangfeld** [comment]
@percona-csalguero @rnovikovP @rasika-chivate @denisok Is anyone able to explain to me what Percona's intended release cycle for this project is? I am [9 months into asking when a release is going to be cut](https://github.com/percona/percona-toolkit/pull/504#pullrequestreview-764209642) to fix a [really simple little bug](https://github.com/percona/percona-toolkit/pull/474). 9 months and counting to release that little fix does not count as a functional release process.

**Comment by denisok** [comment]
@benlangfeld I wish I can explain, it always has some strange priority for us. I would bring this on today for discussions with PMs and also would discuss it in engineering to figure out what should be a process for release for this.
CC @dygksy

File: Changelog [hunk 10]
```diff
@@ -1,21 +1,28 @@
...
+ * Fixed bug   PT-1336: pt-stalk removes user's files from the destination directory
+ * Fixed bug   PT-1398: pt-stalk gets the incorrect mysqld pid when the host installed a multi MySQL instance
  * Improvement PT-1417: Inconsistent creation of toolkit tables
  * Fixed bug   PT-1627:	pt-mysql-summary doesn't verify which version of jemalloc is in use
  * Fixed bug   PT-1747: pt-online-schema-change: metadata lock can break database for rebuild_constraints
- * Improvement PT-1800: PTDEBUG=1 exposes passwords
- * Fixed bug   PT-1900:	pt-query-digest not hiding parameter properly sometimes when parameter=binary
- * Improvement PT-1940:	ptsoc dropswap with mysql8: revise rejection (Thanks duxthefux)
+ * Improvement PT-1800: The environment variable PTDEBUG=1 exposes the passwords
+ * Improvement PT-1887: pt-diskstat is not working for new kernels
```
Comment (LenzGr) [inline Changelog]: PT-1887 was actually a bug fix, not an improvement.

Comment (percona-csalguero) [reply Changelog]: :+1:

**Review by rnovikovP — APPROVED** [review]
